docs/contributing/hacking.rst

```diff
@@ -72,49 +72,31 @@ It is also straightforward to build Go from source:
 Once you can compile to ``linux/amd64``, you should be able to compile Deis'
 components as normal.
 
-Configuring Boot2Docker (Mac)
+Configuring Docker Toolbox and Machine (Mac)
 `````````````````````````````
 
 Deis needs a Docker registry running independently of the Deis cluster. On
-OS X, you will need to run this docker registry inside of Boot2Docker (http://boot2docker.io).
+OS X, you will need Docker Toolbox (https://www.docker.com/toolbox) to run
+the registry inside of a VirtualBox image.
 
-Install Boot2Docker according to the normal installation instructions. When you
-run ``init``, we highly recommend allocating a large disk, since the Docker
-registry that will live there is fairly large.
-
-.. code-block:: console
-
-    $ boot2docker init --disksize=100000
-
-That will create virtual disk that can eventually take up a full 100,000MB of
-disk space. Then start up Boot2Docker.
-
-Once you have run ``boot2docker up``, you should be able to connect to it. You
-need to make a minor editor to the boot2docker config:
-
-.. code-block:: console
+.. note::
 
-    $ boot2docker ip
-    192.168.59.103
-    $ boot2docker ssh sudo vi /var/lib/boot2docker/profile
+    Previously, Deis used boot2docker to run the registry. However, Docker has
+    deprecated boot2docker in favor of Docker Toolbox.
 
-Inside of the profile, you need to add one line, making sure to set the IP
-address to whatever ``boot2docker ip`` printed.
+Install Docker Toolbox according to the normal installation instructions. Then
+create a new image for hosting your Deis Docker registry:
 
 .. code-block:: console
 
-    EXTRA_ARGS="--insecure-registry 192.168.59.103:5000"
-
-Once that line has been added, you can either restart boot2docker's docker
-server, or you can restart boot2docker. We recommend the latter.
-
-.. code-block:: console
+    $ docker-machine create --driver virtualbox --virtualbox-disk-size=100000 --engine-insecure-registry=192.168.59.103:5000 deis-registry
```

Comment:
How do you know the ip address of the virtual machine? It is allocated by virtualbox. For example, my docker machine has the ip of `192.168.99.101`.